mxcubeweb/core/components/harvester.py

```diff
@@ -0,0 +1,505 @@
+                "phiy": md.phiyMotor.get_value(),
+                "phiz": md.phizMotor.get_value(),
+                "centring_focus": md.centringFocus.get_value(),
+                "centring_vertical": md.centringVertical.get_value(),
+            }
+
+            saved_position = md.saved_motor_position
+            # find offset position based on old and new motor position
+            new_motor_offset = {
+                "focus": motor_pos_dict["focus"] - saved_position["focus"],
+                "phiy": motor_pos_dict["phiy"] - saved_position["phiy"],
+                "phiz": motor_pos_dict["phiz"] - saved_position["phiz"],
+                "centring_focus": (
+                    motor_pos_dict["centring_focus"] - saved_position["centring_focus"]
+                ),
+                "centring_vertical": (
+                    motor_pos_dict["centring_vertical"]
+                    - saved_position["centring_vertical"]
+                ),
+            }
+
+            calibrated_motor_offset = {
+                "focus": new_motor_offset["focus"] + new_motor_offset["centring_focus"],
+                "phiy": new_motor_offset["phiy"],
+                "phiz": (
+                    new_motor_offset["phiz"] + new_motor_offset["centring_vertical"]
+                ),
+            }
+
+            print(calibrated_motor_offset)
+            # we store the motor offset in the Harvester, to be used for sample centering
+            harvester_device.store_calibrated_pin(
+                calibrated_motor_offset["focus"],
+                calibrated_motor_offset["phiy"],
+                calibrated_motor_offset["phiz"],
+            )
+
+            harvester_device.set_calibrate_state(False)
+        except Exception as ex:
+            logging.getLogger("user_level_log").exception(
+                f"Pin Calibration / validation Failed {str(ex)}"
+            )
+            return False
+
+        return True
+
+    def get_sample_info(self, location: str) -> dict[str]:
+        samples_list = HWR.beamline.sample_changer.get_sample_list()
+        sample_data = {}
+        for s in samples_list:
+            if s.get_address() == location or s.get_id() == location:
+                sample_data = {
+                    "location": s.get_address(),
+                    "sampleID": s.get_address(),
+                    "crystalUUID": s.get_id(),
+                    "sampleName": s.get_name(),
+                }
+                return sample_data
+
+        return sample_data
+
+    def harvest_and_mount_sample(self, xtal_uuid: str) -> dict[str, bool, int]:
+        try:
+            harvester_device = HWR.beamline.harvester
+            self.harvest_crystal(xtal_uuid)
+            harvester_device._wait_sample_transfer_ready(None)
+
+            sample = self.get_sample_info(xtal_uuid)
+            self.app.sample_changer.mount_sample(sample)
+        except Exception:
+            return "Could not Harvest Crystal"
```

Comment:
Maybe log the exception at least, and then maybe someone will be able to fix the issue if any shows up.